pkgs/servers/rainloop/default.nix

```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchurl, unzip, dataPath ? "/etc/rainloop",
+  edition ? "", hash ? "bf558752ce5035dc036702878f453fa4ba3da68c25e77271a079ac285d8acf76" }: stdenv.mkDerivation rec {
+  name = "rainloop-${edition}${version}";
+  version = "1.12.0";
+
+  buildInputs = [ unzip ];
+
+  src = fetchurl {
+    url = "https://github.com/RainLoop/rainloop-webmail/releases/download/v${version}/rainloop-${edition}${version}.zip";
+    sha256 = hash;
+  };
+
+  unpackPhase = ''
+    unzip -q $src
+  '';
+
+  installPhase = ''
+    mkdir $out
+    cp -r * $out
+
+    rm -rf $out/env-vars $out/data
+    ln -s ${dataPath} $out/data
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Simple, modern & fast web-based email client";
+    homepage = "https://www.rainloop.net";
+    downloadPage = "https://github.com/RainLoop/rainloop-webmail/releases";
+    license = licenses.agpl3;
+    platforms = platforms.all;
+    maintainers = with maintainers; [ das_j ];
+  };
```

Comment:
Can you add a `common` function like here? https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/sql/postgresql/default.nix